### PR TITLE
Fix excessive copying when querying the size of a KTX backed texture

### PR DESCRIPTION
**Pull request by jherico**
`Texture::accessStoredMipFace` returns an in-RAM copy of texture data for the specified mip / face.  However, if the texture backing is a KTX file, then it must copy the memory mapped file data from disk into memory.  This means that calls to `Texture::accessStoredMipFace` should be avoided whenever possible, especially on the main thread, as they may trigger blocking IO.

This PR modifies the `Size Texture::getStoredMipSize(uint16 level) const` method, which is in turn depended on by the `Size Texture::getStoredSize() const` method to no longer access the mip data itself just to calculate its size, instead relying on `Texture::getStoredMipFaceSize` which has no IO impact.  

## Testing

Application behavior should be unchanged.  Please verify that scenes with textures still load _twice_ (once to verify loading from the network, then restart the application and load the scene again to verify they load properly from the now populated KTX cache)

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 175298a6d53e85514b70410710f3f71da8cdc6e2 available at: https://gist.github.com/516dc66f54fd87d2b4ecffb164ab04f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10256-175298a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10256-175298a.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 24a1a7cfbde72cba888685910a9a3756644d8a1d available at: https://gist.github.com/516dc66f54fd87d2b4ecffb164ab04f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10256-24a1a7c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10256-24a1a7c.dmg

**Comment by samcake** [comment]
TEsted myself on i5 + amd
with success

**Comment by hifi-gustavo** [comment]
Build console output for commit: fa0fb113293d9847da10089727504f9186c99e10 available at: https://gist.github.com/516dc66f54fd87d2b4ecffb164ab04f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10256-fa0fb11.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10256-fa0fb11.dmg

**Comment by OmegaHeron** [comment]
Testing

**Comment by OmegaHeron** [comment]
Tested Windows 10
QA Pass

Loaded dev-playa, dev-welcome with cold cache, restarted interface and ran through each with hot cache.
Looks same as master for both, loaded all models/textures from cold/hot cache and general smoke test = ok.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7be40a48a19b099fe5017702e8b0eab1f7dc662b available at: https://gist.github.com/516dc66f54fd87d2b4ecffb164ab04f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10256-7be40a4.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10256-7be40a4.dmg

File: libraries/gpu-gl/src/gpu/gl/GLTexture.cpp [hunk 1]
```diff
@@ -207,38 +207,32 @@ TransferJob::TransferJob(const GLTexture& parent, uint16_t sourceMip, uint16_t t
...
-    } else {
+    // If we're copying a subsection of the mip, do additional calculations to find the size and offset of the segment
+    if (0 != lines) {
         transferDimensions.y = lines;
         auto dimensions = _parent._gpuObject.evalMipDimensions(sourceMip);
-        auto bytesPerLine = (uint32_t)mipSize / dimensions.y;
-        auto sourceOffset = bytesPerLine * lineOffset;
+        auto bytesPerLine = (uint32_t)_transferSize / dimensions.y;
+        _transferOffset = bytesPerLine * lineOffset;
         _transferSize = bytesPerLine * lines;
```
Comment (jherico) [inline libraries/gpu-gl/src/gpu/gl/GLTexture.cpp]: yes, the default size is the full mip size, but if 0 != lines, we need to recalculate the size

**Pull request merged**